### PR TITLE
Integration test prepare job

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

Add an integration test verifying that the prepare job returns 0 (non error exit code).

The work includes enables a number of things in the containerized pyspark environment including connecting to SQL Servers, creating databases, migrating from .NET project code, reading from and writing to Delta tables, mocking HTTP endpoints.

## References

* #616

### CODE REVIEW

File: source/databricks/tests/integration/conftest.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        "database_name": master_data_database_name,
+        "sa_user_id": sa_user_id,
+        "sa_user_pass": sa_user_pass
+    }
+
+
+@pytest.fixture(scope="session")
+def master_data_database(source_path):
+    # Create database if not exists
+    # conn = pyodbc.connect('DRIVER={/opt/microsoft/msodbcsql18/lib64/libmsodbcsql-18.0.so.1.1};'
```
Comment (kristianschneider) [inline source/databricks/tests/integration/conftest.py:76]: Perhaps remove comment?

**Review by lasrinnil — APPROVED** [review]

File: source/databricks/geh_stream/schemas/time_series_points_schema.py [hunk 2]
```diff
@@ -12,12 +12,17 @@
...
       StructField(Colname.quantity, DecimalType(18, 3), False),
       StructField(Colname.quality, StringType(), False),
       StructField(Colname.time, TimestampType(), False),
+      StructField(Colname.year, IntegerType(), False),
+      StructField(Colname.month, IntegerType(), False),
+      StructField(Colname.day, IntegerType(), False),
       StructField(Colname.registration_time, TimestampType(), False),
 ])
+
+time_series_points_schema.__doc__ = "This schema must conform to the schema used by the domain publishing the time series points."
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply source/databricks/geh_stream/schemas/time_series_points_schema.py:28]: Agree. It'll be done in an upcoming PR. The problem is even bigger between the time-series and aggregations domain.

File: source/databricks/geh_stream/shared/data_loader.py [hunk 3]
```diff
@@ -27,27 +27,22 @@
...
-    if args_dict.get('data_storage_account_name') is not None:
-        spark_conf.set(f'fs.azure.account.key.{args.data_storage_account_name}.dfs.core.windows.net', args.data_storage_account_key)
-
-    if args_dict.get('shared_storage_account_name') is not None:
-        spark_conf.set(f'fs.azure.account.key.{args.shared_storage_account_name}.dfs.core.windows.net', args.shared_storage_account_key)
+    spark_conf = (SparkConf(loadDefaults=True)
+                  .set("spark.sql.session.timeZone", "UTC")
+                  .set("spark.databricks.io.cache.enabled", "True")
+                  .set("spark.databricks.delta.formatCheck.enabled", "False")
+                  .set(f'fs.azure.account.key.{args.shared_storage_account_name}.dfs.core.windows.net', args.shared_storage_account_key)
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply source/databricks/geh_stream/shared/data_loader.py:35]: Thx. Removed.

File: source/databricks/geh_stream/shared/data_loader.py [hunk 4]
```diff
@@ -70,7 +65,7 @@ def __load_from_sql_table(spark: SparkSession, args: Namespace, table_name: str)
             .load())
 
 
-def load_metering_points(args: Namespace, spark: SparkSession, grid_areas: List[str]) -> DataFrame:
+def load_metering_points(beginning_date_time, end_date_time, args: Namespace, spark: SparkSession, grid_areas: List[str]) -> DataFrame:
     df = (__load_from_sql_table(spark, args, "MeteringPoint")
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply source/databricks/geh_stream/shared/data_loader.py:69]: Good catch! Will fix in this PR.

File: source/databricks/tests/integration/conftest.py (hunk 1 above)
Comment (BjarkeMeier) [reply source/databricks/tests/integration/conftest.py:76]: Done.

File: source/databricks/geh_stream/shared/data_loader.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (kristianschneider) [reply source/databricks/geh_stream/shared/data_loader.py:68]: Super. You can resolve this comment at your convenience

**Review by kristianschneider — APPROVED** [review]

File: source/databricks/geh_stream/schemas/time_series_points_schema.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply source/databricks/geh_stream/schemas/time_series_points_schema.py:28]: As per team talk this morning I no longer consider this part of the tasks.
Let me know if you want some action anyway.

File: source/databricks/geh_stream/shared/data_loader.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply source/databricks/geh_stream/shared/data_loader.py:68]: It doesn't. Or well, yes it does but shouldn't. The problem is actually related to the implementation of the `filter_on_period` function.

The current (in main) implementation (reformatted for readability):
```python
def filter_on_period(df: DataFrame, period: Period) -> DataFrame:
    return (df
            .filter(col(Colname.from_date).cast("long") < period.to_date.timestamp())
            .filter(col(Colname.to_date).cast("long") > period.from_date.timestamp()))
```

causes this SQL query:

```sql
exec sp_executesql N'SELECT "RowId","MeteringPointId","GridArea","EnergySupplierId","IsGridLoss","IsSystemCorrection","FromDate","ToDate"
FROM GridLossSysCorr WHERE ("FromDate" IS NOT NULL) AND ("ToDate" IS NOT NULL)'
```

While this fixed implementation:
```python
def filter_on_period(df: DataFrame, period: Period) -> DataFrame:
    return (df
            .filter(col(Colname.from_date) < period.to_date)
            .filter(col(Colname.to_date) > period.from_date))
```

results in this SQL query:
```sql
exec sp_executesql N'
SELECT "RowId","MeteringPointId","GridArea","EnergySupplierId","IsGridLoss","IsSystemCorrection","FromDate","ToDate" 
FROM GridLossSysCorr WHERE ("FromDate" IS NOT NULL) AND ("ToDate" IS NOT NULL) AND ("FromDate" < ''2020-01-01 00:00:00.0'') AND ("ToDate" > ''2020-01-01 00:00:00.0'')'
```

**Review by kristianschneider — APPROVED** [review]

File: source/databricks/geh_stream/schemas/time_series_points_schema.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lasrinnil) [reply source/databricks/geh_stream/schemas/time_series_points_schema.py:28]: If it becomes an issue down the line, lets fix it then :)

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**